src/snapshot/download.js

```
@@ -0,0 +1,66 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var toImage = require('../plot_api/to_image');
+var Lib = require('../lib');
+
+/**
+ * @param {object} gd figure Object
+ * @param {object} opts option object
+ * @param opts.format 'jpeg' | 'png' | 'webp' | 'svg'
+ */
+function downloadImage(gd, opts) {
+    
+    // check for undefined opts
+    opts = opts || {};
+    
+    // default to png
+    opts.format = opts.format || 'png';
+    
+    if(Lib.isIE()) {
+        Lib.notifier('Snapshotting is unavailable in Internet Explorer. ' +
+                     'Consider exporting your images using the Plotly Cloud', 'long');
+        return;
+    }
+
+    if(gd._snapshotInProgress) {
+        Lib.notifier('Snapshotting is still in progress - please hold', 'long');
```

Similarly, `Plotly.downloadImage` should throw an Error here.